python/gps/algorithm/algorithm_traj_opt.py

```diff
@@ -98,15 +105,107 @@ def _stepadjust(self, m):
 
         self._set_new_mult(predicted_impr, actual_impr, m)
 
+    def _update_cost(self):
+        """ Update the cost objective in each iteration. """
```

Comment:
It would be best to test the importance weight code numerically against matlab, which can be done by saving out the input data in mat files, load via numpy's mat file reader, transpose them correctly, and pass them through this code, seeing if you get the same result.